sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DoNotHardcodeCredentials_CustomValues.cs

```diff
@@ -34,4 +34,12 @@ public void StandardAPI(SecureString secureString, string nonHardcodedPassword,
             new PasswordDeriveBytes("hardcoded", byteArray, cspParams); // Noncompliant
         }
     }
+
+    public class FalseNegatives
+    {
+        public void Method()
+        {
+            string x = @"\*+?|}{][)(^$.# =something"; // FN
```

Comment:
It's the `\b` at the start of the regex. Fixing this will trigger some false positives so I prefer to keep it like this for now. I think it's a good compromise since this is really a corner case.

Reply:
I don't want to change the regex. It's just not clear why it is a FN. So adding some letter `x` at the beginning of the config value will work this around?
It would be nice to document this as FN with `*` config value like `string y = @"*=something"; // FN keyword must start with letter`